source/features/comments-time-machine-links.tsx

```diff
@@ -9,10 +9,10 @@ function init() {
 
 	for (const comment of comments) {
 		const timestampEl = select('relative-time', comment);
-		const timestamp = timestampEl.attributes.datetime.value;
+		const timestamp = timestampEl.attributes['datetime'].value; // eslint-disable-line dot-notation
```

Comment:
I disabled the `dot-notation` rule here because the type checking fails otherwise. I know it works in JS land, but the types are strict, so it's required to use `[]` notation.

Reply:
I think the correct way would be to cast `select<HTMLTimeElement>` or whatever element GitHub’s `relative-time` extends

Reply:
You should cast it to the correct element instead, after the “select” call.

Reply:
Looks like even when it is cast to `HTMLTimeElement`, it still complains.

![image](https://user-images.githubusercontent.com/833231/52166107-17c32580-26d7-11e9-9a05-0f45986ea976.png)

It's actually the `<relative-time />` web component from [github.com/github/time-elements](https://github.com/github/time-elements) that they're using, so you have no choice but to access it via `attributes` with the `[]` notation.

![image](https://user-images.githubusercontent.com/833231/52166419-f19f8480-26da-11e9-85de-fa1935505086.png)

Reply:
Maybe the issue is that `time` actually has a `dateTime` attribute: https://developer.mozilla.org/en-US/docs/Web/API/HTMLTimeElement (camelcase)

```js
const timestampEl = select<HTMLTimeElement>('relative-time', comment);
const timestamp = timestampEl.attributes.dateTime.value;
```

Does that work?

Reply:
Context: https://github.com/Microsoft/TypeScript/blob/9d3707d671592d030386956c9ce39e539b8d0972/src/lib/dom.generated.d.ts#L4757

I give up 🤷‍♂️